pandas/core/frame.py

```diff
@@ -5838,7 +5838,12 @@ class    max    type
                         level_values, lab, allow_fill=True, fill_value=lev._na_value
                     )
 
-                new_obj.insert(0, name, level_values)
+                new_obj.insert(
+                    0,
+                    name,
+                    level_values,
+                    allow_duplicates=self.flags.allows_duplicate_labels,
+                )
```

Comment:
This would probably need a deprecation warning, and should be a separate PR.

Reply:
I had commented in your "Some refinements" PR: https://github.com/johnzangwill/pandas/pull/5#issuecomment-985468199

I will make a separate PR for this.